### PR TITLE
add some decimate_pro arguments

### PR DESCRIPTION
**Pull request by darikg**
Add support for boundary vertex deletion and max vertex degree parameters

### CODE REVIEW

**Comment by akaszynski** [comment]
@tkoyama010, do you know why the triage workflow is still failing?

**Comment by tkoyama010** [comment]
@akaszynski In this case, the reason is this PR doesn't have any labels.

**Comment by akaszynski** [comment]
> @akaszynski In this case, the reason is this PR doesn't have any labels.

Suggesting resolving this with #2877

File: pyvista/core/filters/poly_data.py [hunk 1]
```diff
@@ -927,6 +929,18 @@ def decimate_pro(
...
+        boundary_vertex_deletion : bool, optional
+            Allow deletion of vertices on the boundary of the mesh.
+            Defaults to ``True``. Turning this off may limit the
+            maximum reduction that may be achieved.
+
+        max_degree : int, optional
+            The maximum vertex degree. If the number of triangles
+            connected to a vertex exceeds ``max_degree``, then the
+            vertex will be split. The complexity of the triangulation
+            algorithm is proportional to degree^2. Setting ``max_degree``
```
Comment (akaszynski) [inline pyvista/core/filters/poly_data.py:941]:
```suggestion
            algorithm is proportional to `max_degree**2`. Setting ``max_degree``
```

**Review by akaszynski — APPROVED** [review]
LGTM. Thanks!

**Review by banesullivan — APPROVED** [review]

**Pull request merged**